jdbc-core/src/test/java/com/salesforce/datacloud/jdbc/hyper/HyperConnectionAdapter.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2024, Salesforce, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.salesforce.datacloud.jdbc.hyper;
+
+import com.google.common.collect.ImmutableMap;
+import com.salesforce.datacloud.hyper.HyperServerConfig;
+import com.salesforce.datacloud.hyper.HyperServerProcess;
+import com.salesforce.datacloud.jdbc.core.DataCloudConnection;
+import com.salesforce.datacloud.jdbc.core.DataCloudConnectionString;
+import com.salesforce.datacloud.jdbc.core.DataCloudJdbcManagedChannel;
+import com.salesforce.datacloud.jdbc.core.HyperGrpcClientExecutor;
+import com.salesforce.datacloud.jdbc.util.DirectDataCloudConnection;
+import io.grpc.ManagedChannelBuilder;
+import java.time.Duration;
+import java.util.Map;
+import java.util.Properties;
+import lombok.AllArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import lombok.val;
+
+@AllArgsConstructor
+@Slf4j
+public class HyperConnectionAdapter implements AutoCloseable {
```

Comment:
why do we need this class now? Which purpose does it serve? The name `HyperConnectionAdapter` sounds as if it would adopt one type of connections to some other type of connection, but this is apparently not what it is doing. The absence of a comment describing when / how to use this class doesn't help, either